python/cudf/cudf/core/column/column.py

```diff
@@ -1186,12 +1186,6 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             data = data.astype(dtype)
     # TODO: Remove nvstrings here when nvstrings is fully removed
     elif isinstance(arbitrary, nvstrings.nvstrings):
```

Comment:
This will error whenever the `as_column` function is called currently instead of only when an nvstrings object is passed. Can we use some type of attribute to check this instead? I.e. something like `arbitrary.__name__ == "nvstrings"`

Reply:
@kkraus14 then we have to check if it has __name__ attribute before we can use that, else it will fail for `list` and other types

Reply:
We could also do `if str(type(arbitrary)) == "nvstrings"` or something like that. As long as we don't have to import `nvstrings` to make it work.

Additionally, we should move this condition below the other typical conditions so we can short circuit in the typical cases earlier.